whitenoise/asgi.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+
+class AsgiWhiteNoise:
+    # This is the same block size as wsgiref.FileWrapper
+    BLOCK_SIZE = 8192
+
+    def __init__(self, whitenoise, application):
+        self.whitenoise = whitenoise
+        self.application = application
```

Comment:
Wrapping the WSGI `WhiteNoise` class for ASGI requires instantiating it with a dummy WSGI application. It also requires the back-and-forth translation of headers. 

I would rather we used subclassing. We could make a `BaseWhiteNoise` that implements all the generic behaviour, with `WhiteNoise` as the WSGI subclass and `AsgiWhiteNoise` as the ASGI subclass.